### ISSUE TITLE
Crash when opening app to Embedded Social fragment

### ISSUE THREAD

**Issue by acrown-msft** [OneBusAway/onebusaway-android#953]
**Summary:** 

If you close the app while the HomeActivity is displaying an Embedded Social fragment, the next app launch results in a NullPointerException and crashes.

**Steps to reproduce:** 

1) Open the app
2) Sign in to Embedded Social
3) Open an Embedded Social fragment from the Navigation Drawer ("My Pins", "Activity Feed", or "My Profile")
4) Kill the application via the app switcher
5) Open OneBusAway from the home screen

**Expected behavior:** 

Not crash

**Observed behavior:** 

Crash

**Device and Android version:** 

Emulated Nexus 5X running Android API 26.
OBA commit 28b3c96054b75cf86c808c69c91c9a8108453d5f

Stacktrace:
```
2018-12-18 14:36:57.435 9528-9528/com.joulespersecond.seattlebusbot E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.joulespersecond.seattlebusbot, PID: 9528
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.joulespersecond.seattlebusbot/org.onebusaway.android.ui.HomeActivity}: android.view.InflateException: Binary XML file line #148: Binary XML file line #148: Error inflating class fragment
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2817)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2892)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1593)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
     Caused by: android.view.InflateException: Binary XML file line #148: Binary XML file line #148: Error inflating class fragment
     Caused by: android.view.InflateException: Binary XML file line #148: Error inflating class fragment
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.microsoft.embeddedsocial.sdk.Options.getBaseTheme()' on a null object reference
        at com.microsoft.embeddedsocial.ui.fragment.base.BaseTabsFragment.<init>(BaseTabsFragment.java:32)
        at com.microsoft.embeddedsocial.ui.fragment.base.BaseProfileFragment.<init>(BaseProfileFragment.java:40)
        at com.microsoft.embeddedsocial.ui.fragment.MyProfileFragment.<init>(MyProfileFragment.java:14)
        at org.onebusaway.android.ui.HomeActivity.showMyProfileFragment(HomeActivity.java:824)
        at org.onebusaway.android.ui.HomeActivity.goToNavDrawerItem(HomeActivity.java:523)
        at org.onebusaway.android.ui.HomeActivity.onNavigationDrawerItemSelected(HomeActivity.java:467)
        at org.onebusaway.android.ui.NavigationDrawerFragment.selectItem(NavigationDrawerFragment.java:317)
        at org.onebusaway.android.ui.NavigationDrawerFragment.onCreate(NavigationDrawerFragment.java:209)
        at androidx.fragment.app.Fragment.performCreate(Fragment.java:2414)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1418)
        at androidx.fragment.app.FragmentManagerImpl.moveToState(FragmentManager.java:1684)
        at androidx.fragment.app.FragmentManagerImpl.addFragment(FragmentManager.java:1930)
        at androidx.fragment.app.FragmentManagerImpl.onCreateView(FragmentManager.java:3745)
        at androidx.fragment.app.FragmentController.onCreateView(FragmentController.java:120)
        at androidx.fragment.app.FragmentActivity.dispatchFragmentsOnCreateView(FragmentActivity.java:405)
        at androidx.fragment.app.FragmentActivity.onCreateView(FragmentActivity.java:387)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:780)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:374)
        at androidx.appcompat.app.AppCompatDelegateImpl.setContentView(AppCompatDelegateImpl.java:469)
        at androidx.appcompat.app.AppCompatActivity.setContentView(AppCompatActivity.java:140)
        at org.onebusaway.android.ui.HomeActivity.onCreate(HomeActivity.java:355)
        at android.app.Activity.performCreate(Activity.java:6975)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1213)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2770)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2892)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1593)
2018-12-18 14:36:57.435 9528-9528/com.joulespersecond.seattlebusbot E/AndroidRuntime:     at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6541)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
```

The key issue looks like the NPE stemming from a null Options class.  Is it possible EmbeddedSocial.init is not called in this flow?

**Comment by barbeau** [comment]
Looks like this is an issue with the workaround of not running ES SDK init in the background implemented in https://github.com/OneBusAway/onebusaway-android/pull/940.

In #940, we postpone ES SDK init until something (Activity or Fragment) enters the onStart() portion of the app lifecycle, indicating that it's running in the foreground instead of the background.  We do this by listening for an onStart() event from an application-wide ProcessLifecycleOwner Observer.  This implementation handles all initialization across the entire app in one place.

However, from the above it looks like ES SDK init() needs to happen prior to onStart() in the ES fragments, and even prior to those Fragment onCreateView() methods because in OBA we don't have control over the `com.microsoft.embeddedsocial.ui.fragment.MyProfileFragment` lifecycle.  So that means it needs to be triggered earlier, for example in HomeActivity.onCreate() or in NavigationDrawerFragment.  The downside of this design is that we will have ES SDK init checking code scattered in several places, but I don't think we can avoid it right now.  I'll take a closer look at implementing a fix.

@acrown-msft Is there anywhere else in OBA where you think this would be an issue (ES SDK init() not called before something hits `onStart()`)?  In other words, if I implement a check for ES SDK init in HomeActivity.onCreate() do you think this would cover all cases?

**Comment by acrown-msft** [comment]
I believe that should be sufficient.  I can't think of a case where an ES fragment is loaded before HomeActivity.onStart() nor are any processes started in the ES SDK before the call to init().
